File: src/js/libro.js
/**
 * La clase representa un libro con propiedades como título, autor, ISBN y estado.
 *  
 * @class
 */
class Libro {
    /**
     * Función constructora que crea objetos de libros.
     * 
     * @param {string} titulo - El parámetro "titulo" representa el título del libro.
     * @param {string} autor - El parámetro "autor" representa el autor del libro.
     * @param {string} ISBN - El ISBN (Número Internacional Normalizado del Libro) es un identificador único para los libros. Se utiliza para identificar una edición específica de un libro.
     * @param {string} estado - El parámetro "estado" representa la condición o estado del libro. Puede usarse para indicar si el libro es nuevo, usado, dañado u otra condición relevante.
     */
    constructor(titulo, autor, ISBN, estado) {
      this.titulo = titulo;
      this.autor = autor;
      this.ISBN = ISBN;
      this.estado = estado;
    }
  
    /**
     * Obtiene el título del libro.
     * @returns {string} El título del libro.
     */
    getTitulo() {
      return this.titulo;
    }

    /**
    * Establece el título del libro.
    * @param {string} titulo - El nuevo título del libro.
    */
    setTitulo(titulo) {
      this.titulo = titulo;
    }

    /**
    * Obtiene el autor del libro.
    * @returns {string} El autor del libro.
    */
    getAutor() {
      return this.autor;
    }

    /**
    * Establece el autor del libro.
    * @param {string} autor - El nuevo autor del libro.
    */
    setAutor(autor) {
      this.autor = autor;
    }

    /**
    * Obtiene el ISBN del libro.
    * @returns {string} El ISBN del libro.
    */
    getISBN() {
      return this.ISBN;
    }

    /**
    * Establece el ISBN del libro.
    * @param {string} ISBN - El nuevo ISBN del libro.
    */
    setISBN(ISBN) {
      this.ISBN = ISBN;
    }

    /**
    * Obtiene el estado del libro.
    * @returns {string} El estado del libro.
    */
    getEstado() {
      return this.estado;
    }

    /**
    * Establece el estado del libro.
    * @param {string} estado - El nuevo estado del libro.
    */
    setEstado(estado) {
      this.estado = estado;
    }

  }
  
  module.exports = Libro;
  

File: src/java/Libro.java
package java;

import java.util.List;
/**
 * La clase Libro posee variables para poder introducir el titulo, autor, ISBN y su estado
 * Posee metodos para listar los libros, ademas de su constructor
 * 
 * 
 * @author Javier Moreno Salas
 * @version 1.0
 */
public class Libro {
    private String titulo;
    private String autor;
    private String ISBN;
    private EstadoLibro estado;
    
    public enum EstadoLibro {
        DISPONIBLE,
        RESERVADO
    }

    /**
     * Constructor para inicializar libro con los valores de titulo, autor y ISBN
     * 
     * @param titulo - El nombre del libro
     * @param autor - EL nombre del autor del libro
     * @param ISBN - El codigo ISBN
     */
    public Libro(String titulo, String autor, String ISBN) {
        this.titulo = titulo;
        this.autor = autor;
        this.ISBN = ISBN;
        this.estado = EstadoLibro.DISPONIBLE; 
    }
    
    /**
     * Metodo el cual muestra por pantalla un listado de libros
     * 
     * @param libros - Lista de libros
     */
    public static void listarLibros(List<Libro> libros) {
        System.out.println("Lista de Libros:");

        for (Libro libro : libros) {
            String estado = (libro.getEstado() == EstadoLibro.DISPONIBLE) ? "Disponible" : "Reservado";
            System.out.println("Título: " + libro.getTitulo());
            System.out.println("Autor: " + libro.getAutor());
            System.out.println("ISBN: " + libro.getISBN());
            System.out.println("Estado: " + estado);
            System.out.println("------------------------");
        }
    }

    /**
     * Obtiene el titulo del Libro
     * 
     * @return el titulo del libro
     */
    public String getTitulo() {
		return titulo;
	}


    /**
     * Establece el titulo
     * 
     * @param titulo - Nuevo titulo del libro
     */
	public void setTitulo(String titulo) {
		this.titulo = titulo;
	}


    /**
     * Obtiene el nombre del autor del Libro
     * 
     * @return el nombre del autor del libro
     */
	public String getAutor() {
		return autor;
	}


    /**
     * Establece el nombre del autor del libro
     * 
     * @param autor - Nuevo nombre del autor del libro
     */
	public void setAutor(String autor) {
		this.autor = autor;
	}


    /**
     * Obtiene el código ISBN del Libro
     * 
     * @return el código ISBN del libro
     */
	public String getISBN() {
		return ISBN;
	}


    /**
     * Establece el código ISBN del libro
     * 
     * @param iSBN - Nuevo código ISBN del libro
     */
	public void setISBN(String iSBN) {
		ISBN = iSBN;
	}


    /**
     * Obtiene el estado del Libro
     * 
     * @return el estado del libro
     */
	public EstadoLibro getEstado() {
        return estado;
    }

    /**
     * Establece el estado del libro
     * 
     * @param estado - Nuevo estado del libro
     */
    public void setEstado(EstadoLibro estado) {
        this.estado = estado;
    }

    /**
     * Genera una representación en formato de cadena (String) del objeto Libro.
     * 
     * @return una cadena que representa el libro en formato de texto
     */
    @Override
    public String toString() {
        return "Libro{" +
                "titulo='" + getTitulo() + '\'' +
                ", autor='" + getAutor() + '\'' +
                ", ISBN='" + getISBN() + '\'' +
                ", estado=" + getEstado() +
                '}';
    }

}
